lib/auth/apiserver.go

```diff
@@ -2145,7 +2145,15 @@ func (s *APIServer) getRole(auth ClientI, w http.ResponseWriter, r *http.Request
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return rawMessage(services.MarshalRole(role, services.WithVersion(version), services.PreserveResourceID()))
+	roleV4, ok := role.(*types.RoleV4)
+	if !ok {
+		return nil, trace.BadParameter("Unrecognized role version")
+	}
+	downgraded, err := downgradeRole(context.Background(), roleV4)
```

Comment:
Can you add a comment block here with what version this migration code can be removed? Search for `DELETE IN` comments in the code base and follow that pattern. Will make it easy to remove migration logic in the future.

Reply:
All of the apiserver role endpoints have `DELETE IN 7.0` where they are registered, I've added comments at the function definitions as well https://github.com/gravitational/teleport/blob/6684b13672cc5706d687bb9001b40e158476f447/lib/auth/apiserver.go#L194-L199